ci/disallow_hash_comments_in_spec_files.sh

```diff
@@ -0,0 +1,12 @@
+#! /bin/bash
+
+set -euo pipefail
+
+# shellcheck disable=SC1091
+source ci/ci_helpers.sh
+
+# We do not use comments starting with # in spec files because it creates warnings from
+# preprocessor that expects directives after this character.
+
+# `# ` -> `-- `
+find src/test/regress/spec/*.spec -print0 | xargs -0 sed -i 's!# !// !g'
```

Comment:
what about ?

```suggestion
find src/test/regress/spec/*.spec -print0 | xargs -0 sed -i 's!#!//!g'
```

Reply:
this regex would also capture `#define` clauses. I would not want the following to happen:

```diff
-#include "isolation_mx_common.include.spec"
+//include "isolation_mx_common.include.spec"
```